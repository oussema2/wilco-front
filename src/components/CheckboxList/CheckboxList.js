import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { observer } from 'mobx-react-lite';
import { styles } from './CheckboxList.styles';
import { checkbox, checkboxUnselected } from '../../assets/icons';
import SelectMultiple from '../SelectMultiple/SelectMultiple';

const CheckboxList = ( {
	testID, items, selectedItems, onSelectionsChange, ParentComponent
} ) => (
	<View testID={testID}>
		<SelectMultiple
			items={items}
			rowStyle={styles.row}
			checkboxSource={checkboxUnselected}
			selectedCheckboxSource={checkbox}
			checkboxStyle={styles.checkbox}
			style={styles.selectMultiple}
			labelStyle={styles.label}
			selectedLabelStyle={styles.selectedLabel}
			selectedItems={selectedItems}
			onSelectionsChange={onSelectionsChange}
			selectedCheckboxStyle={styles.selectedCheckbox}
			ParentComponent={ParentComponent}
		/>
	</View>
);

CheckboxList.propTypes = {
	testID: PropTypes.string,
	items: PropTypes.arrayOf( PropTypes.any ).isRequired,
	selectedItems: PropTypes.arrayOf( PropTypes.any ).isRequired,
	onSelectionsChange: PropTypes.func.isRequired,
	ParentComponent: PropTypes.elementType
};

CheckboxList.defaultProps = {
	testID: 'CheckboxList-Component',
	ParentComponent: View
};

export default observer( CheckboxList );
