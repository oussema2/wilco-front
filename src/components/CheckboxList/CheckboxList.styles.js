import { StyleSheet } from 'react-native';
import { fonts, palette } from '../../Theme';

export const styles = StyleSheet.create( {
	row: {
		marginVertical: 5, backgroundColor: 'transparent', marginHorizontal: 0, paddingHorizontal: 0, borderWidth: 0, borderBottomWidth: 0
	},
	checkbox: {
		borderWidth: 0,
		backgroundColor: palette.grayscale.mercury,
		borderRadius: 4,
		width: 20,
		height: 20,
		marginRight: 12
	},
	selectMultiple: { borderWidth: 0, borderBottomWidth: 0 },
	selectedCheckbox: { backgroundColor: 'transparent' },
	label: {
		...fonts.body
	},
	selectedLabel: {
		...fonts.bodyFocus
	}
} );
