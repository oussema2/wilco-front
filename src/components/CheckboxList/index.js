export { default as Comment } from './CheckboxList';
