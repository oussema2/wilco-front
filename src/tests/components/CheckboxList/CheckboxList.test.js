import * as React from 'react';
import { render } from '@testing-library/react-native';
import CheckboxList from '../../../components/CheckboxList/CheckboxList';

describe( 'CheckboxList', () => {
	describe( 'with the default props', () => {
		it( 'renders the CheckboxList component correctly', async () => {
			const list = [ 'Item 1', 'Item 2' ];
			const selectedItems = [ 'Item 1', 'Item 2' ];
			const component = render( <CheckboxList
				items={list}
				onSelectionsChange={() => { }}
				testID="testID"
				selectedItems={selectedItems}
			/> );
			expect( component.queryByTestId( 'testID' ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );
} );
