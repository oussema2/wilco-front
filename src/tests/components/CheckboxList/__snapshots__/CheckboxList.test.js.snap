// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxList with the default props renders the CheckboxList component correctly 1`] = `
<View
  testID="testID"
>
  <View
    testID="select-multiple-testID"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="row-Item 1-testID"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#cccccc",
              "borderBottomWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            Object {
              "backgroundColor": "transparent",
              "borderBottomWidth": 0,
              "borderWidth": 0,
              "marginHorizontal": 0,
              "marginVertical": 5,
              "paddingHorizontal": 0,
            },
            null,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/icons/checkbox.png",
            }
          }
          style={
            Array [
              Object {
                "height": 24,
                "marginRight": 5,
                "width": 24,
              },
              Object {
                "backgroundColor": "#E3E3E3",
                "borderRadius": 4,
                "borderWidth": 0,
                "height": 20,
                "marginRight": 12,
                "width": 20,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {},
                Object {
                  "fontFamily": "NotoSansJP-Regular",
                  "fontSize": 14,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 20,
                },
                Object {
                  "fontFamily": "NotoSansJP-Bold",
                  "fontSize": 14,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Item 1
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="row-Item 2-testID"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#cccccc",
              "borderBottomWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "padding": 15,
            },
            Object {
              "backgroundColor": "transparent",
              "borderBottomWidth": 0,
              "borderWidth": 0,
              "marginHorizontal": 0,
              "marginVertical": 5,
              "paddingHorizontal": 0,
            },
            null,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/icons/checkbox.png",
            }
          }
          style={
            Array [
              Object {
                "height": 24,
                "marginRight": 5,
                "width": 24,
              },
              Object {
                "backgroundColor": "#E3E3E3",
                "borderRadius": 4,
                "borderWidth": 0,
                "height": 20,
                "marginRight": 12,
                "width": 20,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {},
                Object {
                  "fontFamily": "NotoSansJP-Regular",
                  "fontSize": 14,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 20,
                },
                Object {
                  "fontFamily": "NotoSansJP-Bold",
                  "fontSize": 14,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Item 2
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
